### PR TITLE
dolphin-emu: remove stable, 5.0-17995 -> 5.0-18498

### PR DESCRIPTION
**Pull request by SuperSamus**
###### Description of changes
Update Dolphin Emulator to the latest beta.
Removed the stable version, since the developers have long recommended not to use it. Thus, renamed `dolphin-emu-beta` to `dolphin-emu`.
Adjusted some of the dependencies.
I have been a little aggressive in removing some Darwin-specific stuff. I don't use Darwin, so I'll see what the CI says about it.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/emulators/dolphin-emu/default.nix [hunk 1]
```diff
@@ -1,126 +1,191 @@
...
-    glib
-    gettext
-    libpthreadstubs
-    libXrandr
-    libXext
-    readline
-    openal
+    fmt_8
+    hidapi
     libevdev
```
Comment (tpwrules) [inline pkgs/applications/emulators/dolphin-emu/default.nix:81]: This is a duplicate from the Linux-only section and prevents evaluation on Darwin where `libevdev` is unsupported.

File: pkgs/applications/emulators/dolphin-emu/default.nix [hunk 2]
```diff
@@ -1,126 +1,191 @@
...
     libusb1
-    libpulseaudio
-    wxGTK30
-    soundtouch
-    miniupnpc
+    libXdmcp
     mbedtls_2
-    curl
-    lzo
+    mgba
```
Comment (tpwrules) [inline pkgs/applications/emulators/dolphin-emu/default.nix:89]: Prevents evaluation on Darwin where `mgba` is unsupported (but should probably be marked broken?).

**Comment by tpwrules** [comment]
Builds and plays my (you'll scarcely believe) legitimate backups on `aarch64-darwin`.

The Vulkan plugin doesn't load because the Qt wrap hook wraps the MoltenVK .dylib and breaks it. I'm not sure how best to fix it, it's probably ultimately a bug in the hook. We can undo the hook's action or just be okay with the fact that users will need to run on Metal.

**Review by sugar700 — APPROVED** [review]
I think removing stable version at this point makes sense, it's an ancient release, nobody should be using it, and at this point pretty much every other Linux distribution (Arch, Debian, Fedora, Gentoo, GUIX, Mageia, openSUSE, Solus, Ubuntu, Flathub packages) only provides a development version of Dolphin. Once Dolphin 6.0 gets released we may want to start to distribute stable versions by default, but in current state there is no reason to maintain old version.

As for the change itself, it works, and it's neat that Dolphin can detect external mGBA now :).

**Pull request merged**